aws_xray_sdk/core/models/entity.py

```diff
@@ -256,36 +257,42 @@ def get_origin_trace_header(self):
     def serialize(self):
         """
         Serialize to JSON document that can be accepted by the
-        X-Ray backend service. It uses jsonpickle to perform
-        serialization.
+        X-Ray backend service. It uses json to perform serialization.
         """
         try:
-            return jsonpickle.encode(self, unpicklable=False)
+            return json.dumps(self.to_dict(), default=str)
         except Exception:
-            log.exception("got an exception during serialization")
+            log.exception("Failed to serialize %s", self.name)
 
-    def _delete_empty_properties(self, properties):
+    def to_dict(self):
         """
-        Delete empty properties before serialization to avoid
-        extra keys with empty values in the output json.
+        Convert Entity(Segment/Subsegment) object to dict
+        with required properties that have non-empty values. 
         """
-        if not self.parent_id:
-            del properties['parent_id']
-        if not self.subsegments:
-            del properties['subsegments']
-        if not self.aws:
-            del properties['aws']
-        if not self.http:
-            del properties['http']
-        if not self.cause:
-            del properties['cause']
-        if not self.annotations:
-            del properties['annotations']
-        if not self.metadata:
-            del properties['metadata']
-        properties.pop(ORIGIN_TRACE_HEADER_ATTR_KEY, None)
-
-        del properties['sampled']
+        entity_dict = {}
+            
+        for key, value in vars(self).items():
+            if isinstance(value, bool) or value:
+                if key == 'subsegments':
```

Comment:
How do we handle the case if key is `annotations`, `aws` or `http` ? I see that you're handling `metadata` case separately. Shouldn't we do the same with `annotations` too? since in that case also we don't know how many key-value pairs would be there.

Reply:
`annotations`, `aws` or `http` are all `dict` with serializable values like `string`, `int` or `boolean`, so there is no need to handle these cases.